addons/account/models/account_move.py

```diff
@@ -679,27 +679,30 @@ def _compute_suitable_journal_ids(self):
     @api.depends('posted_before', 'state', 'journal_id', 'date')
     def _compute_name(self):
         self = self.sorted(lambda m: (m.date, m.ref or '', m.id))
-        highest_name = self[0]._get_last_sequence(lock=False) if self else False
 
         for move in self:
-            name_not_set = not move.name or move.name == '/'
-            if not highest_name and move == self[0] and not move.posted_before and move.date and name_not_set:
-                # In the form view, we need to compute a default sequence so that the user can edit
-                # it. We only check the first move as an approximation (enough for new in form view)
-                move._set_next_sequence()
-            elif move.quick_edit_mode and not move.posted_before:
-                # We always suggest the next sequence as the default name of the new move
-                if name_not_set or not move._sequence_matches_date():
-                    move._set_next_sequence()
-            elif not move.posted_before and not move._sequence_matches_date():
-                # The date changed before posting on first move of period
-                move._set_next_sequence()
-            elif (name_not_set and move.state == 'posted'):
+            move_has_name = move.name and move.name != '/'
+            if move_has_name or move.state != 'posted':
+                if not move.posted_before and not move._sequence_matches_date():
+                    if move._get_last_sequence(lock=False):
+                        # The name does not match the date and the move is not the first in the period:
+                        # Reset to draft
+                        move.name = '/'
+                        continue
+                else:
+                    if move_has_name and move.posted_before or not move_has_name and move._get_last_sequence(lock=False):
```

Comment:
Wasn't the check on date being set supposed to be put in this PR?

Reply:
Yes, it's been moved to line 699 here, since the logic is a little bit different. There is no `try`-`except` anymore in 16.0, but instead the check with `move._sequence_matches_date()`, so it needed a little bit of adjustment.